.ci/cico_rhche_compatibility_test.sh

```diff
@@ -5,129 +5,181 @@
-	exit 1
-fi
-
-echo "********** Prepare environment for running compatibility test with upstream version of che: $CHE_VERSION **********"
-
-eval "$(./env-toolkit load -f jenkins-env.json -r \
-        ^DEVSHIFT_TAG_LEN$ \
-        ^QUAY_ \
-        ^KEYCLOAK \
-        ^BUILD_NUMBER$ \
-        ^JOB_NAME$ \
-        ^ghprbPullId$ \
-        ^RH_CHE \
-        ^FABRIC8_HUB_TOKEN)"
-        
-source ./config
-source ./.ci/functional_tests_utils.sh
-
-echo "Checking credentials:"
-checkAllCreds
-echo "Installing dependencies:"
-installDependencies
-
-#Check branch for tracking changes/Create new branch
-BRANCH="upstream-check-$CHE_VERSION"
-echo "Checking if branch $BRANCH exists."
-set +e
-git checkout "$BRANCH"
-return_code=$?
-set -e
-if [ $return_code -eq 0 ]; then 
-  echo "Branch $BRANCH found - rebasing."
-  git rebase origin/master "$BRANCH"
-else 
-  echo "Branch $BRANCH not found - creating new one."
-  git checkout -b "$BRANCH"
-	
-  #change version of used che
-  echo ">>> change upstream version to: $CHE_VERSION"
-  scl enable rh-maven33 "mvn versions:update-parent  versions:commit -DallowSnapshots=true -DparentVersion=[${CHE_VERSION}] -U"
-fi
-
-echo "Setting image tags for pushing to quay."
-#Get last commit short hash from upstream che
-longHashUpstream=$(curl -s https://api.github.com/repos/eclipse/che/commits/master | jq .sha)
-shortHashUpstream=${longHashUpstream:1:7}
-
-#Get last commit short hash from rh-che branch 
-longHashDownstream=$(git log | grep -m 1 commit | head -1 | cut -d" " -f 2)
-shortHashDownstream=${longHashDownstream:0:7}
-
-#DOCKER_IMAGE_TAG is used for running tests
-#DOCKER_IMAGE_TAG_WITH_SHORTHASHES is used for tracking changes
-export DOCKER_IMAGE_TAG="upstream-check-latest"	
-export DOCKER_IMAGE_TAG_WITH_SHORTHASHES="upstream-check-$shortHashUpstream-$shortHashDownstream"
-export PROJECT_NAMESPACE=compatibility-check
-
-#set values needed for checking/creating PR
-RELATED_PR_TITLE="Update to $(echo $CHE_VERSION | cut -d'-' -f 1)"
-PR_BODY="Tracking changes for fixing compatibility with upstream $CHE_VERSION. This PR was created automatically by Jenkins from job $JOB_NAME"
-PR_HEAD="$BRANCH"
-PR_BASE="master"
-
-PULL_REQUESTS=$(curl -s https://api.github.com/repos/redhat-developer/rh-che/pulls?state=open | jq '.[].title')
-
-echo "Checking if PR exists."
-PR_EXISTS=1
-while read -r pr_title
-do
-  if [[ "$pr_title" == "\"$RELATED_PR_TITLE\"" ]]; then
-    echo "Pull request for tracking changes of version $CHE_VERSION has been already created."
-    PR_EXISTS=0
-    break
+set -e -m +x
+
+function checkPullRequest() {
+  PULL_REQUESTS=$(curl -s https://api.github.com/repos/redhat-developer/rh-che/pulls?state=open | jq '.[].title')
+  PR_EXISTS="false"
+  while read -r pr_title
+  do
+    if [[ "$pr_title" == "\"$RELATED_PR_TITLE\"" ]]; then
+      PR_EXISTS="true"
+      break
+    fi
+  done <<< "$PULL_REQUESTS"
+
+  if [[ "$PR_EXISTS" == "false" ]]; then
+    echo "false"
+  else
+    echo "true"
```

Comment:
This function could be reworked to return an exit code:
```bash
if [ "$PR_EXISTS" == "true"; then
  return 0
else
  return 1
fi
```
It can then be used directly in a condition check:
```bash
if ! checkPullRequest; then
  # PR doesn't exist
fi
```

Reply:
good catch :) I will do a complete refactor of the script at some later time anyways.